commit 0bac6af3c18a989771e2d5bb9199f8aae64943f8
Author: Nitzan Mordechai <nmordech@ibm.com>
Date:   Tue May 19 12:30:59 2026 +0000

    mgr/ClusterState: remove debug 30 memory spike
    
    Remove the full pg_map JSON dump at dout(30) which was flooding the
    log queue with multi-MB entries on every tick. The much smaller
    incremental dump is retained for debugging.
    
    Fixes: https://tracker.ceph.com/issues/64082
    Signed-off-by: Nitzan Mordechai <nmordech@ibm.com>

diff --git a/src/mgr/ClusterState.cc b/src/mgr/ClusterState.cc
index ee24e3203f1..2201caa1776 100644
--- a/src/mgr/ClusterState.cc
+++ b/src/mgr/ClusterState.cc
@@ -142,16 +142,14 @@ void ClusterState::update_delta_stats()
   pending_inc.version = pg_map.version + 1; // to make apply_incremental happy
   dout(10) << " v" << pending_inc.version << dendl;
 
-  dout(30) << " pg_map before:\n";
-  JSONFormatter jf(true);
-  jf.dump_object("pg_map", pg_map);
-  jf.flush(*_dout);
-  *_dout << dendl;
-  dout(30) << " incremental:\n";
-  JSONFormatter jf(true);
-  jf.dump_object("pending_inc", pending_inc);
-  jf.flush(*_dout);
-  *_dout << dendl;
+  if (!pending_inc.empty()) {
+    dout(30) << " incremental:\n";
+    JSONFormatter jf(true);
+    jf.dump_object("pending_inc", pending_inc);
+    jf.flush(*_dout);
+    *_dout << dendl;
+  }
+
   pg_map.apply_incremental(g_ceph_context, pending_inc);
   pending_inc = PGMap::Incremental();
 }
@@ -179,16 +177,13 @@ void ClusterState::notify_osdmap(const OSDMap &osd_map)
   PGMapUpdater::check_down_pgs(osd_map, pg_map, true,
 			       need_check_down_pg_osds, &pending_inc);
 
-  dout(30) << " pg_map before:\n";
-  JSONFormatter jf(true);
-  jf.dump_object("pg_map", pg_map);
-  jf.flush(*_dout);
-  *_dout << dendl;
-  dout(30) << " incremental:\n";
-  JSONFormatter jf(true);
-  jf.dump_object("pending_inc", pending_inc);
-  jf.flush(*_dout);
-  *_dout << dendl;
+  if (!pending_inc.empty()) {
+    dout(30) << " incremental:\n";
+    JSONFormatter jf(true);
+    jf.dump_object("pending_inc", pending_inc);
+    jf.flush(*_dout);
+    *_dout << dendl;
+  }
 
   pg_map.apply_incremental(g_ceph_context, pending_inc);
   pending_inc = PGMap::Incremental();
